pandas/core/frame.py

```diff
@@ -4068,6 +4069,12 @@ def _apply_standard(self, func, axis, ignore_failures=False, reduce=True):
                 self._is_datelike_mixed_type):
             reduce = False
 
+        # we cannot reduce using non-numpy dtypes,
+        # as demonstrated in gh-12244
+        if (reduce and ExtensionDtype.is_dtype(
+                self.values.dtype)):
+            reduce = False
```

Comment:
move this into the block below (which only activates ``if reduce``).

the reason is that a mixed dtype frame could trigger ``.values`` twice